control-plane/connect-inject/webhook_v2/consul_dataplane_sidecar.go

```diff
@@ -0,0 +1,474 @@
+			}
+		}
+		// Ensure that none of the user's containers have the same UID as consul-dataplane. At this point in injection the meshWebhook
+		// has only injected init containers so all containers defined in pod.Spec.Containers are from the user.
+		for _, c := range pod.Spec.Containers {
+			// User container and consul-dataplane container cannot have the same UID.
+			if c.SecurityContext != nil && c.SecurityContext.RunAsUser != nil && *c.SecurityContext.RunAsUser == sidecarUserAndGroupID && c.Image != w.ImageConsulDataplane {
+				return corev1.Container{}, fmt.Errorf("container %q has runAsUser set to the same UID \"%d\" as consul-dataplane which is not allowed", c.Name, sidecarUserAndGroupID)
+			}
+		}
+		container.SecurityContext = &corev1.SecurityContext{
+			RunAsUser:              pointer.Int64(sidecarUserAndGroupID),
+			RunAsGroup:             pointer.Int64(sidecarUserAndGroupID),
+			RunAsNonRoot:           pointer.Bool(true),
+			ReadOnlyRootFilesystem: pointer.Bool(true),
+		}
+	}
+
+	return container, nil
+}
+
+func (w *MeshWebhook) getContainerSidecarArgs(namespace corev1.Namespace, bearerTokenFile string, pod corev1.Pod) ([]string, error) {
+	proxyIDFileName := "/consul/mesh-inject/proxyid"
+
+	envoyConcurrency := w.DefaultEnvoyProxyConcurrency
+
+	// Check to see if the user has overriden concurrency via an annotation.
+	if envoyConcurrencyAnnotation, ok := pod.Annotations[constants.AnnotationEnvoyProxyConcurrency]; ok {
+		val, err := strconv.ParseUint(envoyConcurrencyAnnotation, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse annotation %q: %w", constants.AnnotationEnvoyProxyConcurrency, err)
+		}
+		envoyConcurrency = int(val)
+	}
+
+	args := []string{
+		"-addresses", w.ConsulAddress,
+		"-grpc-port=" + strconv.Itoa(w.ConsulConfig.GRPCPort),
+		"-proxy-service-id-path=" + proxyIDFileName,
+		"-log-level=" + w.LogLevel,
+		"-log-json=" + strconv.FormatBool(w.LogJSON),
+		"-envoy-concurrency=" + strconv.Itoa(envoyConcurrency),
+	}
+
+	if w.SkipServerWatch {
+		args = append(args, "-server-watch-disabled=true")
+	}
+
+	if w.AuthMethod != "" {
+		args = append(args,
+			"-credential-type=login",
+			"-login-auth-method="+w.AuthMethod,
+			"-login-bearer-token-path="+bearerTokenFile,
+			// We don't know the pod name at this time, so we must use environment variables to populate the login-meta instead.
+		)
+		if w.EnableNamespaces {
+			if w.EnableK8SNSMirroring {
+				args = append(args, "-login-namespace=default")
+			} else {
+				args = append(args, "-login-namespace="+w.consulNamespace(namespace.Name))
+			}
+		}
+		if w.ConsulPartition != "" {
+			args = append(args, "-login-partition="+w.ConsulPartition)
+		}
+	}
+	if w.EnableNamespaces {
+		args = append(args, "-service-namespace="+w.consulNamespace(namespace.Name))
+	}
+	if w.ConsulPartition != "" {
+		args = append(args, "-service-partition="+w.ConsulPartition)
+	}
```

Comment:
We'll need to use the new `-proxy-blah` args here from Iryna's PR: https://github.com/hashicorp/consul-dataplane/pull/242/files#diff-36c1ea42620072b194bdd2aebc8225b33766ae2b2fc526d0f2bcd2c1640fd093R71-R74